src/helpers/documentation.js

```diff
@@ -235,9 +209,85 @@ export const buildDocsContents = (data, rootDir) => {
     insertPageData(relPath, contents, articles, rootPageData);
   });
 
+  // Now that content is in proper order, travel through tree and add Next Up link to each article
```

Comment:
my mistake before was adding the next up links before we properly grouped all the data. That's why things were in the right order in the left nav, but the next up links for nested items did not work at all